### PR TITLE
feat(watt): add dropdown to Watt Design System

### PR DESCRIPTION
**Pull request by dzhavat**
### Watt Design System
- Add `watt-dropdown` component
- Styling adjustments to `watt-form-field` component
- Showcase and documentation in Storybook

Single selection

image

Multi-selection

image

## References

- #338

### CODE REVIEW

File: libs/ui-watt/src/lib/components/dropdown/+storybook/storybook-dropdown-template-driven-forms.stories.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+ 
+ <watt-dropdown [(ngModel)]="singleSelectionModel" [options]="options"></watt-dropdown>
+ 
+ 5. Wrap the dropdown component in a "watt-form-field"
+
+ <watt-form-field>
+  <watt-dropdown [(ngModel)]="singleSelectionModel" [options]="options"></watt-dropdown>
+ </watt-form-field>`;
+
+export const singleSelect: Story<WattDropdownComponent> = (
```
Comment (frontend-specialisten) [inline libs/ui-watt/src/lib/components/dropdown/+storybook/storybook-dropdown-template-driven-forms.stories.ts:78]: Question: Should we have some showcases for `large` form-fields as well or is this out of scope for now?

File: libs/ui-watt/src/lib/components/dropdown/watt-dropdown.component.ts [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+@Component({
+  selector: 'watt-dropdown',
+  templateUrl: './watt-dropdown.component.html',
+  styleUrls: ['./watt-dropdown.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class WattDropdownComponent
+  implements ControlValueAccessor, OnInit, OnDestroy
+{
+  /** Subject that emits when the component has been destroyed. */
```
Comment (frontend-specialisten) [inline libs/ui-watt/src/lib/components/dropdown/watt-dropdown.component.ts:56]: Suggestion: Let's figure out an convention how we want our comments formatted, so they can be more consistent.

**Review by frontend-specialisten — APPROVED** [review]
Praise: Looking good! Just have a single question and a nitpick, I'm approving this anyways since these changes won't break anything for the users, so I don't see any reasons for holding this back so ex. Titans can start using it. My comments can easily be implemented in other pull-request(s) but ofc. it's up to you.

File: libs/ui-watt/src/lib/components/dropdown/watt-dropdown.component.ts (hunk 2 above)
Comment (dzhavat) [reply libs/ui-watt/src/lib/components/dropdown/watt-dropdown.component.ts:56]: Good point. I'll then suggest `//` for comments that explain a property, code chunk, etc. and TypeDoc kind of comments for API artifacts (property on a component, interface, type, etc.). What do you think?

Will also delete the comment above. We use this pattern in many places and we should be familiar with it by now.

File: libs/ui-watt/src/lib/components/dropdown/+storybook/storybook-dropdown-template-driven-forms.stories.ts (hunk 1 above)
Comment (dzhavat) [reply libs/ui-watt/src/lib/components/dropdown/+storybook/storybook-dropdown-template-driven-forms.stories.ts:78]: I think we should include it. It's also a good opportunity to test how the dropdown will work in a `large` form-field. Looks like Titans are only using `normal` form-fields for now. So we can make a follow-up PR with a showcase for `large` form-fields.

Comment (frontend-specialisten) [reply libs/ui-watt/src/lib/components/dropdown/+storybook/storybook-dropdown-template-driven-forms.stories.ts:78]: Totally fine by me, do you create an issue for this follow-up for the backlog? :)

**Review by frontend-specialisten — APPROVED** [review]

**Review by frontend-specialisten — APPROVED** [review]

File: libs/ui-watt/src/lib/components/dropdown/+storybook/storybook-dropdown-template-driven-forms.stories.ts (hunk 1 above)
Comment (dzhavat) [reply libs/ui-watt/src/lib/components/dropdown/+storybook/storybook-dropdown-template-driven-forms.stories.ts:78]: Just created the issue in the backlog.

**Pull request merged**